arch/arm/core/cortex_m/prep_c.c

```diff
@@ -33,12 +33,18 @@ _GENERIC_SECTION(.vt_pointer_section) void *_vector_table_pointer;
 #define VECTOR_ADDRESS 0
 void __weak relocate_vector_table(void)
 {
-#if defined(CONFIG_XIP) && (CONFIG_FLASH_BASE_ADDRESS != 0) || \
+#if defined(CONFIG_SW_VECTOR_RELAY)
+	_vector_table_pointer = _vector_start;
+#elif defined(CONFIG_XIP) && (CONFIG_FLASH_BASE_ADDRESS != 0) || \
     !defined(CONFIG_XIP) && (CONFIG_SRAM_BASE_ADDRESS != 0)
+#if defined(CONFIG_CPU_CORTEX_M0_HAS_VECTOR_TABLE_REMAP)
```

Comment:
I'm not sure this is going to work. Vector Table Remap doesn't mean it has VTOR, but rather a mechanism to remap its table. In particular I think the STM32F0 does something different than VTOR for this. @b0661, @simple2017 can you weigh in here?

Reply:
The other option is that you redefine the weak function `relocate_vector_table` in your `soc.c` like the STM32F0 code has done, in order to have your own particular version that uses VTOR. I think that might be the simplest.

Reply:
[Here](https://github.com/zephyrproject-rtos/zephyr/commit/2f7688bfd3d3114ff039b5b3d1795a9388d6f355#diff-ee625053bbccbb0d77aba679bdf6a18f) is the relevant STM32F0 commit if you want to take a look.

Reply:
I think we should merge [this PR](https://github.com/zephyrproject-rtos/zephyr/pull/5988) and you could then simply set `CPU_CORTEX_M_HAS_VTOR` for this IC.

Reply:
I don't think this gonna work on STM32F0, check [here](https://github.com/zephyrproject-rtos/zephyr/blob/master/arch/arm/soc/st_stm32/stm32f0/soc.c#L39) as @carlescufi mentioned. Its not really a offset register but a remap address 0x2000_0000 to address 0x0000_0000

Reply:
So I will remove the changes of `prep_c.c` from this PR and add CPU_CORTEX_M_HAS_VTOR for LPC54414's M0+ core instead. Once [the other PR](https://github.com/zephyrproject-rtos/zephyr/pull/5988) is merged.

Reply:
Yep. The PR is merged now.